ckanext/datastore/backend/postgres.py

```diff
@@ -1969,13 +2000,19 @@ def delete_data(context: Context, data_dict: dict[str, Any]):
                                              fields_types, query_dict)
 
     where_clause, where_values = _where(query_dict['where'])
-    sql_string = u'DELETE FROM "{0}" {1}'.format(
-        data_dict['resource_id'],
-        where_clause
+    # (canada fork only): https://github.com/ckan/ckan/pull/8684
+    sql_string = u'DELETE FROM {0} {1} {2}'.format(
+        identifier(data_dict['resource_id']),
+        where_clause,
+        'RETURNING {return_columns}'.format(
+            return_columns=return_columns) if data_dict['include_records'] else ''
     )
 
     try:
-        _execute_single_statement(context, sql_string, where_values)
+        # (canada fork only): https://github.com/ckan/ckan/pull/8684
+        results =_execute_single_statement(context, sql_string, where_values)
+        if data_dict['include_records']:
+            data_dict['deleted_records'] = [dict(r) for r in results.mappings().all()]
```

Comment:
I could see this causing a MemoryError when deleting all the rows in a large table.

For dsaudit we only need the first N rows to populate the activity. Can we apply a LIMIT to a DELETE ... RETURNING?